drivers/docker/handle.go

```diff
@@ -159,7 +159,7 @@ func (h *taskHandle) Kill(killTimeout time.Duration, signal string) error {
 	// Signal is used to kill the container with the desired signal before
 	// calling StopContainer
 	if signal == "" {
-		err = h.client.StopContainer(h.containerID, uint(killTimeout.Seconds()))
+		err = h.waitClient.StopContainer(h.containerID, uint(killTimeout.Seconds()))
```

Comment:
TBH, I am not sure of all the implications of this, but I can confirm that this does "solve" the issue in the ticket.